File: src/main/java/DahRealPanda/plugins/untitled1/colony/model/BuildingInfo.java
package DahRealPanda.plugins.untitled1.colony.model;

import java.util.ArrayList;
import java.util.List;

/**
 * A colony building and (when applicable) the resources required to build/upgrade it.
 */
public class BuildingInfo {
    public int id;
    public String name;
    public String type;
    public String kind = "building"; // "building" or "decoration"
    public int level;
    public int x;
    public int y;
    public int z;

    public boolean beingBuilt;
    public int workOrderId = -1;

    public List<ResourceEntry> required = new ArrayList<>();
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/model/BuildingInfo.kt
package DahRealPanda.plugins.colonyweb.model

/**
 * A colony building and (when applicable) the resources required to build/upgrade it.
 */
data class BuildingInfo(
    @JvmField var id: Int = 0,
    @JvmField var name: String = "",
    @JvmField var type: String = "",
    @JvmField var kind: String = "building", // "building" or "decoration"
    @JvmField var blockId: String? = null,   // registry id of the MineColonies hut block (icon source)
    @JvmField var level: Int = 0,
    @JvmField var x: Int = 0,
    @JvmField var y: Int = 0,
    @JvmField var z: Int = 0,
    @JvmField var beingBuilt: Boolean = false,
    @JvmField var workOrderId: Int = -1,
    @JvmField var required: MutableList<ResourceEntry> = ArrayList()
)
